### ISSUE TITLE
Progressbar issue

### ISSUE THREAD

**Issue by hwoarangmy** [OpenDungeons/OpenDungeons#895]
While playing with the progressbar, I've found a weird issue. I've tried to fill it with hardcoded values to see how it behaves and got unexpected results:
- When I fill it with 0.25, I get no visual at all (like if it was 0%)
- When I fill it with 0.33, I get a small visual (a bit less than 20%)
- When I fill it with 0.5, it is half filled as expected
- When I fill it with 0.75, it is fully filled (like if it was 100%)

Note that I think I had done the same tests with your PR and I think it used to work. Also note that I recently upgraded to cegui 0.8.4 which might be the reason why.
However, I've no 0.8.3 to test :-(

@Bertram25 @akien-mga  do you have the issue ?

**Comment by akien-mga** [comment]
I've been running cegui 0.8.4 for many months now. I'll see if I can reproduce the issue.

**Comment by Bertram25** [comment]
@hwoarangmy Does that issue apply both on spells and research buttons? (with cegui 0.8.4)

Personally, I've been using cegui 0.8.3/0.8.4 and even 0.8.5-to-be versions but I couldn't see any behavioral difference with my latest PR.
I'll try yours asap.

Feel also free to see whether a progressbar param is getting the way or missing:
http://static.cegui.org.uk/static/TaharezLookProperties.html#ProgressBar like `StepSize` or `ReversedProgress`. Who knows?

**Comment by hwoarangmy** [comment]
> Does that issue apply both on spells and research buttons? (with cegui 0.8.4)

yes. That's the first thing I've tried. I will have a look tonight and maybe try with cegui 0.8.3 if you tell me that you don't have the issue

**Comment by hwoarangmy** [comment]
I've had a look at the progressbars we use. I've found something a bit weird about the area as the area is defined bigger than the button:
https://github.com/OpenDungeons/OpenDungeons/blob/development/gui/WindowResearchTree.layout#L20

I tried to play with it and set it to 0,0 to 1,1. As expected, horizontally, the bar was smaller than the button (which explains the hack I guess). However, vertically, the size is ok. So, I guess that the bug comes from the progressbar height being too big and that explains what I've seen.

I've made a patch and will upload it once my current PR will be in. Version 0.6.0 is close ^^
